airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/DestinationTaskLauncher.kt

```diff
@@ -277,10 +280,13 @@ class DefaultDestinationTaskLauncher(
             }
 
             if (streamManager.isBatchProcessingComplete()) {
-                log.info { "Batch processing complete: Starting close stream task for $stream" }
-
-                val task = closeStreamTaskFactory.make(this, stream)
-                enqueue(task)
+                if (closeStreamHasRun.getOrPut(stream) { AtomicBoolean(false) }.setOnce()) {
```

Comment:
Nit: maybe pre-populate the map so we don't have to reason about the guarantees of `getOrPut` and can just think about the atomic boolean.